osutil/io.go

```diff
@@ -226,3 +226,64 @@ func AtomicWriteChown(filename string, reader io.Reader, perm os.FileMode, flags
 
 	return aw.Commit()
 }
+
+// AtomicRename attempts to rename a path from oldName to newName atomically.
+func AtomicRename(oldName, newName string) error {
+	var oldDir, newDir *os.File
+
+	if !snapdUnsafeIO {
+		oldDirPath := filepath.Dir(oldName)
+		newDirPath := filepath.Dir(newName)
+
+		o, err := os.Open(oldDirPath)
```

Comment:
I was surprised that there is no need to specify that you intend to open a directory (no explicit O_DIRECTORY) but this is apparently the way to do it in go.

Reply:
Yeah, AtomicFile does the same.